File: client-app/src/app/models/employeeHr.ts
export interface IEmployeeHr {

    employeeNumber: string;

    hrRepresentative: string;
    //hrRepresentative?: number;

    rateType: string;
    //rateType?: number;

    // typeTechnical: string;

    payRate: string;

    halogenUser: boolean;

    hoursWeek: string;

    adpFileNumber: string;

    address: string;

    hrSetupCompleteCbx: boolean;

    welcomeEmailSentDate?: Date;

    phoneOption: string;
    phoneOptionDate?: Date;

    gasCardezPass: string;
    gasCardezPassDate?: Date;

    businessCard: string;
    businessCardDate?: Date;

    orientationDate?: Date;
    orientationDateSetDate?: Date;

    professionalResume: string;
    professionalResumeDate?: Date;

    meetwithMarketing: string;
    welcomeBagSent: string;
    travelForOrientation: string;
    reportOnFirstDayTime: string;

    i9LinkCbx: boolean;
    rvTrainingCompleteCbx: boolean;
    addedToRvCbx: boolean;
    cccCompanyManualCbx: boolean;

    hrOnboardingCompleteCbx: boolean;
}

export class EmployeeHrFormValues implements IEmployeeHr {

    employeeNumber: string = '';


    hrRepresentative: string = '';
    //hrRepresentative?: number;

    rateType: string = '';
    //rateType?: number;

    // typeTechnical: string;

    payRate: string = '';

    halogenUser: boolean = false;

    hoursWeek: string = '';

    adpFileNumber: string = '';

    address: string = '';

    hrSetupCompleteCbx: boolean = false;

    welcomeEmailSentDate?: Date;

    phoneOption: string = '';
    phoneOptionDate?: Date;

    gasCardezPass: string = '';
    gasCardezPassDate?: Date;

    businessCard: string = '';
    businessCardDate?: Date;

    orientationDate?: Date;
    orientationDateSetDate?: Date;

    professionalResume: string = '';
    professionalResumeDate?: Date;

    meetwithMarketing: string = '';
    welcomeBagSent: string = '';
    travelForOrientation: string = '';
    reportOnFirstDayTime: string = '';

    i9LinkCbx: boolean = false;
    rvTrainingCompleteCbx: boolean = false;
    addedToRvCbx: boolean = false;
    cccCompanyManualCbx: boolean = false;

    hrOnboardingCompleteCbx: boolean = false;
}

File: Domain/EmployeeHr.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class EmployeeHr
    {
        [Key]
        public string EmployeeNumber { get; set; }

        [Required]
        public Int16 HrRepresentative { get; set; }

        [Required]
        public Int16 RateType { get; set; }

        [StringLength(20)]
        public string PayRate { get; set; }

        public Boolean HalogenUser { get; set; }

        [StringLength(20)]
        public string HoursWeek { get; set; }

        [StringLength(20)]
        public string AdpFileNumber { get; set; }

        [StringLength(200)]
        public string Address { get; set; }

        public Boolean HrSetupCompleteCbx { get; set; }


        public DateTime? WelcomeEmailSentDate { get; set; }

        public Int16 PhoneOption { get; set; }

        public DateTime? PhoneOptionDate { get; set; }

        public Int16 GasCardezPass { get; set; }
        public DateTime? GasCardezPassDate { get; set; }

        public Int16 BusinessCard { get; set; }
        public DateTime? BusinessCardDate { get; set; }

        public DateTime? OrientationDate { get; set; }
        public DateTime? OrientationDateSetDate { get; set; }

        public Int16 ProfessionalResume { get; set; }
        public DateTime? ProfessionalResumeDate { get; set; }

        [StringLength(80)]
        public string MeetwithMarketing { get; set; }

        [StringLength(80)]
        public string WelcomeBagSent { get; set; }

        [StringLength(80)]
        public string TravelForOrientation { get; set; }

        [StringLength(80)]
        public string ReportOnFirstDayTime { get; set; }

        public Boolean I9LinkCbx { get; set; }
        public Boolean RvTrainingCompleteCbx { get; set; }
        public Boolean AddedToRvCbx { get; set; }
        public Boolean CccCompanyManualCbx { get; set; }

        public Boolean HrOnboardingCompleteCbx { get; set; }
    }
}